### ISSUE TITLE
index is out of bounds error

### ISSUE THREAD

**Issue by DonkeyShot21** [LandingEllipse/SLDTk#32]
Hi, I'm trying to use the tool. I get the following error:

```
sldtk -i hmi_20110105_175555_continuum.png

Traceback (most recent call last):
  File "c:\users\enric\appdata\local\programs\python\python36\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "c:\users\enric\appdata\local\programs\python\python36\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\enric\AppData\Local\Programs\Python\Python36\Scripts\sldtk.exe\__main__.py", line 9, in <module>
  File "c:\users\enric\appdata\local\programs\python\python36\lib\site-packages\sldtk\__init__.py", line 5, in cli
    main()
  File "c:\users\enric\appdata\local\programs\python\python36\lib\site-packages\sldtk\sldtk.py", line 65, in main
    stack = profile.extract_stack(gray, disk_attr, args['slices'])
  File "c:\users\enric\appdata\local\programs\python\python36\lib\site-packages\sldtk\profile.py", line 53, in extract_stack
    stack = img[y_cart, x_cart]
IndexError: index 4000 is out of bounds for axis 1 with size 4000
```

the error is easy to reproduce, just download a fits file from SDO, open it with astropy.io.fits and save it as png

edit: if I change the threshold it works fine. The error message does not help though ahah
edit2: I want to publish my work and I will maybe use your toolkit, how should I cite you?

**Comment by LandingEllipse** [comment]
Hi @DonkeyShot21 (nice nick, hehe),

Thanks for getting in touch! I'm in the middle of a calibration campaign of an instrument this week so things are a bit hectic. I'll be able to look at this properly and get back to you in the next few days. SLDTk was written back before I had my Python software engineering practices honed in fully, so hopefully the issue will turn out to just be a simple edge case oversight.

With regards to citing (GitHub) software I'll dig up the template I use, it might serve as a suggestion.

Talk to you soon!

**Comment by LandingEllipse** [comment]
Hi @DonkeyShot21,

What seems to be happening here is seems to be that the disk detection algorithm isn't able to discriminate between the solar disk and the background, potentially leading it to falsely detect a disk with a radius/origin combination that at some location extends beyond the image canvas. 

If this is indeed the case it's a trivial fix to add in some sanity checks to the properties of the detected disk before attempting to extract the intensity profile. However, I have not been able to reproduce this issue myself; I don't work with SDO data or FITS files in my day-to-day, so while I did manage to wrestle with JSOC for an HMI continuum FITS file, I couldn't get Astropy to play ball and export a PNG without resorting to manually extracting the numpy array and pushing to disk with imageio. This PNG worked fine with SLDTK which makes me wonder if Astropy might perhaps have done some non-standard scaling during export in your case.

imageio/PIL treats float images by

    im = (img - img.min()) / (img.max() - img.min()) * (np.power(2.0, 8) - 1) + 0.499999999
to get them into uint8. I haven't checked what Astropy does (or if it defers the judgement to the likes of imageio/PIL), as I don't know exactly how you are saving the images as PNGs.

TL;DR: Could you please upload a representative image or FITS->PNG code snippet so that I can reproduce this and verify that my proposed fix/workaround will actually work?

---

Regarding how you would cite SLDTk (should it prove valuable to your research), it doesn't have a DOI at the moment, though I just discovered [Zenodo](https://zenodo.org/) so that might change soon. However depending on your reference style the following is generally an acceptable way to cite online repositories / software libraries:

    A Ladegaard, The Solar Limb Darkening Toolkit (SLDTk), (2019), GitHub repository, https://github.com/LandingEllipse/SLDTk
